tests.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+var assert = require('assert');
+var combyne = require('combyne');
+var visitCombyne = require('./index');
+
+describe('visit-combyne', function() {
+  var template;
+
+  describe('if-elseif conditionals', function() {
+    beforeEach(function() {
+      template = combyne('{%if case1%}1{%elsif case2%}2{%else%}3{%endif%}');
+    });
+
+    it('correctly recurses over all nodes in conditional chains', function() {
+      var visitedNodes = {};
+
+      // Count how many nodes of each type are visited
+      visitCombyne(template.tree.nodes, function(node) {
+        if ( ! visitedNodes[node.type] ) { visitedNodes[node.type] = 0; }
+        visitedNodes[node.type]++;
+      });
+
+      assert.equal(visitedNodes.ConditionalExpression, 2,
+        'finds three conditional nodes ("if", "elsif")');
+      assert.equal(visitedNodes.Property, 2,
+        'finds two property nodes, "case1" and "case2"');
+      assert.equal(visitedNodes.Text, 3,
+        'finds three text nodes, "1", "2", "3"');
+      assert.equal(visitedNodes['undefined'], 1,
+        'finds one node with undefined type ("else")');
```

Comment:
This feels weird, but the `.els` node has no "type" property so it feels like this inconsistency is within combyne. Not sure whether we should be explicitly verifying it here, or whether all of this should just be simplified to
```js
var visitedNodes = 0;

visitCombyne(template.tree.nodes, function() {
  visitedNodes++;
});

assert.equal(visitedNodes, 8, 'visits all nodes');
```